File: atcoder/abc_174/repsept.py

def nthNum(nMinusOneNum):
    return nMinusOneNum*10 + 7

def main():
    K = int(input())

    crawler = 7
    found = False
    i = 1

    if (K % 2 ==0) or (K % 5 == 0):
        print(-1)
    else:
        a = [0 for i in range (K+1)]
        a[0] = 7
        for i in range (1, K+1):
            a[i] = (a[i-1]*10 + 7) % K
        a[0] = a[0] % K
        for i in a:
            if i == 0:
                print(a.index(i) + 1)
                found = True
                break
        if not found:
            print(-1)
    
main()

File: atcoder/abc_174/repsept.cpp
#include <iostream> 
#include <cmath>

using namespace std;

long long int nthNum (long long int n);

int main()
{
    int K;
    cin >> K;

    long long int crawler = 7;

    for (int i = 0; i < 100; i ++)
    {
        long long int temp = nthNum(crawler);
        cout << temp << endl;
        crawler = temp;
    }
}

long long int nthNum (long long int nMinusOneNum)
{
    return 10 * nMinusOneNum + 7;
}